AWSIoT/Internal/AWSIoTMQTTClient.h

```diff
@@ -226,14 +227,33 @@
 
  @param topic The topic for publish to.
 
- @param ackCallback the callback for ack if QoS > 0.
+ @param ackCallback the callback for ack if qos == 1 || qos == 2
 
  */
 - (void)publishData:(NSData *)data
                 qos:(UInt8)qos
             onTopic:(NSString *)topic
         ackCallback:(AWSIoTMQTTAckBlock)ackCallback;
 
+/**
+Send MQTT message to specified topic
+
+@param data The data to be sent.
+
+@param qos The qos to use when sending (optional, default 0).
+
+@param topic The topic for publish to.
+
+@param retain Sets the retain flag
+
+@param ackCallback the callback for ack if qos == 1 || qos == 2
+
+*/
+- (void)publishData:(NSData*)data
+                qos:(AWSIoTMQTTQoS)qos
+            onTopic:(NSString*)topic
+             retain:(BOOL)retain
+        ackCallback:(AWSIoTMQTTAckBlock)ackCallback;
```

Comment:
Nullability seems inconsistent with the new APIs above. If qos==0 should this be nullable?